VRCFaceTracking/Library/UnifiedLibManager.cs

```diff
@@ -0,0 +1,296 @@
+        {
+            Logger.Msg("Loading External Module " + dll.FullName);
+
+            Type module;
+            ExtTrackingModule moduleObj;
+            try
+            {
+                // Get the first class that implements ExtTrackingModule
+                module = dll.GetTypes().FirstOrDefault(t => t.IsSubclassOf(typeof(ExtTrackingModule)));
+                moduleObj = (ExtTrackingModule)Activator.CreateInstance(module);
+
+                return moduleObj;
+            }
+            catch (ReflectionTypeLoadException e)
+            {
+                foreach (var loaderException in e.LoaderExceptions)
+                {
+                    Logger.Error("LoaderException: " + loaderException.Message);
+                }
+                Logger.Error("Exception loading " + dll + ". Skipping.");
+            }
+            catch (BadImageFormatException e)
+            {
+                Logger.Error("Encountered a .dll with an invalid format: " + e.Message + ". Skipping...");
+            }
+            catch (TypeLoadException)
+            {
+                Logger.Warning("Module " + dll + " does not implement ExtTrackingModule.");
+            }
+
+            return null;
+        }
+
+        public static List<Assembly> LoadExternalAssemblies(string[] path, bool useAttributes = true)
+        {
+            var returnList = new List<Assembly>();
+            foreach (var dll in path)
+            {
+                try
+                {
+                    Assembly loaded = Assembly.LoadFrom(dll);
+                    foreach(Type type in loaded.GetExportedTypes())
+                        if (type.BaseType == typeof(ExtTrackingModule))
+                        {
+                            Logger.Msg(type.ToString() + " implements ExtTrackingModule.");
+                            returnList.Add(loaded);
+                            continue;
+                        }
+                }
+                catch (FileNotFoundException)
+                {
+                    Logger.Warning(dll + " failed to find file. Skipping.");
+                    continue;
+                }
+                catch (ArgumentNullException)
+                {
+                    Logger.Warning(dll + " Assembly mismatch. Skipping.");
+                    continue;
+                }
+                catch (BadImageFormatException)
+                {
+                    Logger.Warning(dll + " Assembly image not of expected format. Skipping.");
+                    continue;
+                }
+                catch (FileLoadException)
+                {
+                    Logger.Warning(dll + " Assembly could not be loaded. Skipping.");
+                    continue;
+                }
+                catch (Exception e)
+                {
+                    Logger.Warning(e.Message + " Assembly not able to be loaded. Skipping. ");
+                    continue;
+                }
```

Comment:
Feels like this would be the only exception we needed if we printed the type of exception as well as the message. The names are pretty self-explanatory though feel free to leave this as-is if you disagree

Reply:
I agree, these were meant to be more concise messages but definitely having the exception message is probably the most useful for troubleshooting code.

Reply:
Removed the extra catches and kept the messaged exception in https://github.com/benaclejames/VRCFaceTracking/pull/109/commits/6c285efaa0bc322e1f46db9842062a7558e432ae